draft-ietf-quic-transport.md

```diff
@@ -2251,16 +2253,21 @@ transport parameter in the TLS handshake.
 Servers MAY communicate a preferred address of each address family (IPv4 and
 IPv6) to allow clients to pick the one most suited to their network attachment.
 
-Once the handshake is finished, the client SHOULD select one of the two
+Once the handshake is confirmed, the client SHOULD select one of the two
 server's preferred addresses and initiate path validation (see
-{{migrate-validate}}) of that address using the connection ID provided in the
-preferred_address transport parameter.
+{{migrate-validate}}) of that address using an active unused connection ID.
 
 If path validation succeeds, the client SHOULD immediately begin sending all
-future packets to the new server address using the new connection ID and
-discontinue use of the old server address.  If path validation fails, the client
-MUST continue sending all future packets to the server's original IP address.
+future packets to the new server and discontinue use of the old server address.
+If path validation fails, the client MUST continue sending all future packets to
+the server's original IP address.
 
+When the client finishes migrating to the preferred address, it retires the
+connection ID that was used for the original path ({{retiring-cids}}).
+Similarly, a client that continues using the original server address SHOULD
+retire the connection ID provided by the preferred_address transport parameter.
+Retirement of either of these connection IDs notifies the server of the
+address the client has chosen.
```

Comment:
This creates a much stronger inference than I think we ever formally documented, though it is a reasonable one for sure.  If your intent is that the server never see data on the old path, then say that instead:

> Once a client has migrated successfully to a network path that uses a server preferred address, it MUST NOT migrate to a path using the address that was used during connection establishment.

Otherwise, the inference you are giving the server to make here is without any support.  Once you say this, you can drop this text.

Reply:
Yeah though not a "MUST NOT", that is already stated in the previous paragraph. That said, I think it would be better to merge the two paragraphs, as having two paragraphs describing what a client should do went migration succeeds (or fails) is confusing.

PTAL at commit f268ac4.